crates/core/src/dbs/capabilities.rs

```diff
@@ -107,6 +107,63 @@ impl std::str::FromStr for FuncTarget {
 	}
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum ExperimentalTarget {
+	RecordReferences,
+}
+
+impl fmt::Display for ExperimentalTarget {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		match self {
+			Self::RecordReferences => write!(f, "record_references"),
+		}
+	}
+}
+
+impl Target for ExperimentalTarget {
+	fn matches(&self, elem: &ExperimentalTarget) -> bool {
+		self == elem
+	}
+}
+
+impl Target<str> for ExperimentalTarget {
+	fn matches(&self, elem: &str) -> bool {
+		match self {
+			Self::RecordReferences => elem.eq_ignore_ascii_case("record_references"),
+		}
+	}
+}
+
+#[derive(Debug, Clone)]
+pub enum ParseExperimentalTargetError {
+	InvalidName,
+}
+
+impl std::error::Error for ParseExperimentalTargetError {}
+impl fmt::Display for ParseExperimentalTargetError {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		match *self {
+			ParseExperimentalTargetError::InvalidName => {
+				write!(f, "invalid experimental target name")
+			}
+		}
+	}
+}
+
+impl std::str::FromStr for ExperimentalTarget {
+	type Err = ParseExperimentalTargetError;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		match s.trim() {
+			s if s.eq_ignore_ascii_case("record_references") => {
+				Ok(ExperimentalTarget::RecordReferences)
+			}
+			_ => Err(ParseExperimentalTargetError::InvalidName),
+		}
```

Comment:
Nitpick: Match here seems a bit unnecessary, this could just be an if else as there is no pattern matching being done.

Reply:
I'm adding graphql and bearer access here too in a followup PR right after, I don't think it's worth changing

Reply:
Still, those all are in the form of `s if s.eq_ignore_ascii_case(...) =>` right? So your not matching any pattern, so the match is just extra syntax which doesn't really do anything.

Reply:
Anyway it was just a nitpick.